lib/external/io.dart

```diff
@@ -19,15 +21,46 @@ ASTNode funcExit(Runtime runtime, ASTNode self, List<ASTNode> args) {
   exit(exitAST.intVal);
 }
 
+StringNode funcGetCwd(Runtime runtime, ASTNode self, List<ASTNode> args) {
+  return StringNode()..stringValue = Directory.current.absolute.path.toString();
+}
+
 ASTNode funcFree(Runtime runtime, ASTNode self, List<ASTNode> args) {
   final Scope scope = getScope(runtime, self);
 
   args.forEach((arg) {
-    if (arg is! VariableNode)
-      return;
+    if (arg is! VariableNode) return;
 
-    scope.variableDefinitions.removeWhere((varDef) => varDef.variableName == arg.variableName);
+    scope.variableDefinitions
+        .removeWhere((varDef) => varDef.variableName == arg.variableName);
   });
 
   return INITIALIZED_NOOP;
-}
\ No newline at end of file
+}
+
+Future<MapNode> funcSystem(
+    Runtime runtime, ASTNode self, List<ASTNode> args) async {
+  runtimeExpectArgs(
+      args, [ASTType.AST_STRING, ASTType.AST_LIST, ASTType.AST_BOOL]);
+
+  Map<String, dynamic> result = {
+    'errorCode': null,
+    'stdout': null,
+    'stderr': null
+  };
+
+  MapNode res = MapNode();
+  bool shell = args[2].boolVal ?? false;
```

Comment:
arg[2] is required since you've added in the `runtimeExpectArgs` call, you don't need a null check here